app/models/environment_class.rb

```diff
@@ -28,4 +29,10 @@ def self.is_in_any_environment(puppetclass, puppetclass_lookup_key)
   def self.key_in_environment(env, puppetclass, puppetclass_lookup_key)
     EnvironmentClass.where(:environment_id => env, :puppetclass_id => puppetclass, :puppetclass_lookup_key_id => puppetclass_lookup_key).first
   end
+
+  def delete_orphaned_lookup_keys
+    if puppetclass_lookup_key.present? && puppetclass_lookup_key.environment_classes.empty?
```

Comment:
This comment got deleted:
> What's the difference between this and belongs_to :puppetclass_lookup_key, :dependent => :destroy ?

There can be multiple environment_classes connected to the same puppetclass_lookup_key and I don't want the lookup key destroyed when you remove one of those records. I want to make sure it deletes only if this is the last environment class for this key.